### PR TITLE
batik.transcoder in ImageServlet

### PR DESCRIPTION
**Pull request by microArtur**
[https://nanopay.atlassian.net/browse/NP-6185](https://nanopay.atlassian.net/browse/NP-6185)

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
- variables not named correctly, use java standard camelCase
- don't use System.out.println() to log errors. grab the foam logger if possible. also, beware that the exception you might be receiving could have been a rethrow. e.getMessage() in that case (and in many other cases) will not give you the direct cause
- png_ostream looks suspicious in how it's opened, maybe do try-with-resources?

File: src/foam/nanos/servlet/ImageServlet.java [hunk 1]
```diff
@@ -69,6 +76,35 @@ protected void service(HttpServletRequest req, HttpServletResponse resp)
           IOUtils.copy(is, resp.getOutputStream());
           return;
         }
+      }else if ( reqPath.endsWith(".png") ){
```
Comment (jlhughes) [inline src/foam/nanos/servlet/ImageServlet.java]:
```suggestion
      } else if ( reqPath.endsWith(".png") ) {
```
Style guide should be on the foam repo (I think) or the nanopay getting started.

File: src/foam/nanos/servlet/ImageServlet.java [hunk 2]
```diff
@@ -69,6 +78,34 @@ protected void service(HttpServletRequest req, HttpServletResponse resp)
...
```
Comment (jlhughes) [inline src/foam/nanos/servlet/ImageServlet.java]:
```suggestion
      } else if ( reqPath.endsWith(".png") ) {
```

File: src/foam/nanos/servlet/ImageServlet.java [hunk 3]
```diff
@@ -69,6 +78,34 @@ protected void service(HttpServletRequest req, HttpServletResponse resp)
...
+        File srcSVG = new File(cwd + "/" + paths[i] + reqPath.replaceFirst("\\.png", ".svg"));
+        if ( srcSVG.isFile() && srcSVG.canRead() && srcSVG.getCanonicalPath().startsWith(new File(paths[i]).getCanonicalPath()) ) {
+
+          // convert .svg to .png
+          String          svgURIInput   = Paths.get(srcSVG.getPath()).toUri().toURL().toString();
+          TranscoderInput inputSVGImage = new TranscoderInput(svgURIInput);
+          try (OutputStream  PNGOutputStream = new FileOutputStream(src.getPath())) {
```
Comment (ghost) [inline src/foam/nanos/servlet/ImageServlet.java]: should be pngOutputStream

File: src/foam/nanos/servlet/ImageServlet.java [hunk 4]
```diff
@@ -69,6 +78,34 @@ protected void service(HttpServletRequest req, HttpServletResponse resp)
...
+            TranscoderOutput outputPNGImage  = new TranscoderOutput(PNGOutputStream);
+            PNGTranscoder    myConverter     = new PNGTranscoder();
+            myConverter.transcode(inputSVGImage, outputPNGImage);
+            PNGOutputStream.flush();
```
Comment (ghost) [inline src/foam/nanos/servlet/ImageServlet.java]: try/finally on output stream should end after this flush happens as it is not used anymore

File: src/foam/util/Emails/EmailsUtility.java [hunk 5]
```diff
@@ -100,6 +100,8 @@ public static void sendEmailFromTemplate(X x, User user, EmailMessage emailMessa
       return;
     }
 
+    emailMessage.setBody( emailMessage.getBody().replaceAll("\\.svg", ".png") );
```
Comment (minsunny) [inline src/foam/util/Emails/EmailsUtility.java:103]: Feeling it should be before email service being called so before this cts.apply(userX, group, emailMessage, templateArgs); above.

**Review by jlhughes — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**